vignettes/backtesting.Rmd

```diff
@@ -84,384 +74,363 @@ doctor_visits <- pub_covidcast(
+color.
+For example, the snapshot on March 1st, 2021 is aquamarine, and increases to
+slightly over 10.
+Every series is necessarily to the left of the snapshot date (since all known
+values must happen before the snapshot is taken[^4]).
+The grey line overlaying the various snapshots represents the "final
+value", which is just the snapshot at the last version in the archive (the
+`versions_end`).
+
+Comparing with the grey line tells us how much the value at the time of the
+snapshot differs with what was eventually reported.
+The drop in January 2021 in the snapshot on `2021-02-01` was initially reported
+as much steeper than it eventually turned out to be, while in the period after
+that the values were initially reported as higher than they actually were.
+
+Handling data latency is important in both real-time forecasting and retrospective
+forecasting.
+Looking at the very first snapshot, `2020-08-01` (the red dotted
+vertical line), there is a noticeable gap between the forecast date and the end
+of the red time-series to its left.
+In fact, if we take a snapshot and get the last `time_value`,
 
 ```{r}
-arx_args_list()
+doctor_visits |>
+  epix_as_of(as.Date("2020-08-01")) |>
+  pull(time_value) |>
+  max()
 ```
 
-These can be modified as needed, by sending your desired arguments into
-`arx_forecaster(args_list = arx_args_list())`. For now we will use the defaults.
+the last day of data is the 25th, a entire week before `2020-08-01`.
+This can require some effort to work around, especially if the latency is
+variable; see `step_adjust_latency()` for some methods included in this package.
+Much of that functionality is built into `arx_forecaster()` using the parameter
+`adjust_ahead`, which we will use below.
 
-__Note__: We will use a __geo-pooled approach__, where we train the model on
-data from all states and territories combined. This is because the data is quite
-similar across states, and pooling the data can help improve the accuracy of the
-forecasts, while also reducing the susceptibility of the model to noise. In the
-interest of computational speed, we only use the 6 state dataset here, but the
-full archive can be used in the same way and has performed well in the past.
-Implementation-wise, geo-pooling is achieved by not using `group_by(geo_value)`
-prior to `epix_slide()`. In other cases, grouping may be preferrable, so we
-leave it to the user to decide, but flag this modeling decision here.
 
-Let's use the `epix_as_of()` method to generate a snapshot of the archive at the
-last date, and then run the forecaster.
+# Backtesting a simple autoregressive forecaster
 
-```{r}
-# Let's forecast 14 days prior to the last date in the archive, to compare.
-forecast_date <- doctor_visits$versions_end - 14
-# The .versions argument selects only the last version in the archive and
-# produces a forecast only on that date.
-forecasts <- doctor_visits %>%
+One of the most common use cases of `epiprocess::epi_archive()` object is for
```

Comment:
I'd prefer not to call this "the most common use-case". I think (hope?) there's
lots of simpler things that one could do with an archive, so I'd rather not give
the impression that you need to be backtesting forecasters for the functionality
to be useful.

Reply:
Fair enough. I'm debating listing some of those; outlier detection and nowcasting come to mind, but I'm not sure if it would end up being too much of a digression tot talk about anyways.